lint/variables.go

```diff
@@ -0,0 +1,174 @@
+	"__to":            time.Date(2020, 7, 13, 20, 19, 9, 254000000, time.UTC),
+	"__name":          "name",
+	"__org":           42,
+	"__org.name":      "orgname",
+	"__user.id":       42,
+	"__user.login":    "user",
+	"__user.email":    "user@test.com",
+	"timeFilter":      "time > now() - 7d",
+	"__timeFilter":    "time > now() - 7d",
+}
+
+func stringValue(name string, value interface{}, kind, format string) (string, error) {
+	switch val := value.(type) {
+	case int:
+		return strconv.Itoa(val), nil
+	case time.Time:
+		// Implements https://grafana.com/docs/grafana/latest/variables/variable-types/global-variables/#__from-and-__to
+		switch kind {
+		case "date":
+			switch format {
+			case "seconds":
+				return strconv.FormatInt(val.Unix(), 10), nil
+			case "iso":
+				return val.Format(time.RFC3339), nil
+			default:
+				return "", fmt.Errorf("Unsupported momentjs time format: " + format)
+			}
+		default:
+			switch format {
+			case "date":
+				return val.Format(time.RFC3339), nil
+			default:
+				return strconv.FormatInt(val.UnixMilli(), 10), nil
+			}
+		}
+	default:
+		// Use variable name as sample value
+		svalue := fmt.Sprintf("%s", value)
+		// For list types, repeat it 3 times (arbitrary value)
+		svalueList := []string{svalue, svalue, svalue}
+		// Implements https://grafana.com/docs/grafana/latest/variables/advanced-variable-format-options/
+		switch format {
+		case "csv":
+			return strings.Join(svalueList, ","), nil
+		case "doublequote":
+			return "\"" + strings.Join(svalueList, "\",\"") + "\"", nil
+		case "glob":
+			return "{" + strings.Join(svalueList, ",") + "}", nil
+		case "json":
+			data, err := json.Marshal(svalueList)
+			if err != nil {
+				return "", err
+			}
+			return string(data), nil
+		case "lucene":
+			return "(\"" + strings.Join(svalueList, "\" OR \"") + "\")", nil
+		case "percentencode":
+			return url.QueryEscape(strings.Join(svalueList, ",")), nil
+		case "pipe":
+			return strings.Join(svalueList, "|"), nil
+		case "raw":
+			return strings.Join(svalueList, ","), nil
+		case "regex":
+			return strings.Join(svalueList, "|"), nil
+		case "singlequote":
+			return "'" + strings.Join(svalueList, "','") + "'", nil
+		case "sqlstring":
+			return "'" + strings.Join(svalueList, "','") + "'", nil
+		case "text":
+			return strings.Join(svalueList, " + "), nil
+		case "queryparam":
+			values := url.Values{}
+			for _, svalue := range svalueList {
+				values.Add("var-"+name, svalue)
+			}
+			return values.Encode(), nil
+		default:
+			return svalue, nil
+		}
+	}
+}
+
+func variableSampleValue(s string) (string, error) {
+	var name, kind, format string
+	parts := strings.Split(s, ":")
+	switch len(parts) {
+	case 1:
+		// No format
+		name = s
+	case 2:
+		// Could be __from:date, variable:csv, ...
+		name = parts[0]
+		format = parts[1]
+	case 3:
+		// Could be __from:date:iso, ...
+		name = parts[0]
+		kind = parts[1]
+		format = parts[2]
+	}
```

Comment:
I think we need a `default` case that returns an error here.

Reply:
Good one. Added, with a test case.